src/ServiceControl.Infrastructure.Metrics/MetricsReporter.cs

```diff
@@ -0,0 +1,55 @@
+﻿namespace ServiceControl.Infrastructure.Metrics
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    public class MetricsReporter
+    {
+        readonly Metrics metrics;
+        readonly Action<string> printLine;
+        readonly TimeSpan interval;
+        CancellationTokenSource tokenSource;
+        Task task;
+
+        public MetricsReporter(Metrics metrics, Action<string> printLine, TimeSpan interval)
+        {
+            this.metrics = metrics;
+            this.printLine = printLine;
+            this.interval = interval;
+        }
+
+        public void Start()
+        {
+            tokenSource = new CancellationTokenSource();
+            task = Task.Run(async () =>
+            {
+                while (!tokenSource.IsCancellationRequested)
+                {
+                    Print();
+                    await Task.Delay(interval).ConfigureAwait(false);
```

Comment:
should probably accept the cancellation token and try catch for shutdown?

Reply:
There is no cancellation token to pass (from Bootstrapper)

Reply:
But you have `tokenSource` available which has a token right?